src/backend/distributed/planner/query_pushdown_planning_utils.c

```diff
@@ -0,0 +1,1280 @@
+/*-------------------------------------------------------------------------
+ *
+ * query_pushdown_planning_utils.c
```

Comment:
"util" is supposed to mean "utility functions", but it has a tendency to mean "pile of uncategorized stuff". 

What if we name this file `query_pushdown_planing.c` and move all the logic relating to query pushdown in here? including e.g. `SubqueryPushdownMultiNodeTree`?

Reply:
I opened the PR in this way to not to spend too much time on refactoring after talked with Onder. Today, we have discussed the scope and timing with Onder and Metin after your review, and decided to extend it to cover the changes you mentioned. I will update the PR with these changes.

Reply:
Ping: we still see `util` in the name?

I think it'd be nice if you can expand the comment here.
* What are the basic requirements for query pushdown with plain words
* When do we use it